strawberryfields/api/devicespec.py

```diff
@@ -97,17 +98,27 @@ def gate_parameters(self):
 
         return gate_parameters
 
+    def fill_template(self, program):
+        """Fill template with parameter values from a program"""
+        if program.type == "tdm" and program.timebins:
+            self._spec["layout"] = self._spec["layout"].format(
+                target=program.target, tm=program.timebins
+            )
```

Comment:
Good point Antal! 

This should probably be changed to use either `device.target` or `self.target` (i.e. `RemoteEngine.target`). These two should always be the same, since `device.target` is created by creating a `DeviceSpec` with `target=self.target` in `RemoteEngine`. `program.target` should be the same _after_ compilation, but will be `None` before compilation (I think), meaning that the above code would only work for already compiled programs.

I'll update this!